crates/sui-core/src/transaction_orchestrator.rs

```diff
@@ -131,12 +130,7 @@ where
         };
         Self::schedule_txes_in_log(pending_tx_log.clone(), quorum_driver_handler.clone());
 
-        // TODO(fastpath): Plumb in transaction driver percentage from TransactionOrchestrator init instead of env var
-        let td_percentage = env::var("TRANSACTION_DRIVER")
-            .ok()
-            .and_then(|s| s.parse::<u8>().ok())
-            .filter(|&p| p <= 100) // Ensure percentage is between 0-100
-            .unwrap_or(0);
+        let td_percentage = choose_transaction_driver_percentage();
```

Comment:
In `choose_transaction_driver_percentage()`, can you change this condition to match on mainnet config only?
```
if std::env::var(sui_types::digests::SUI_PROTOCOL_CONFIG_CHAIN_OVERRIDE_ENV_VAR_NAME).is_ok() {
...
}
```

Also, since protocol config is available here, set td_percentage to 0 if mysticeti_fastpath() is disabled?